src/main/java/com/minecolonies/core/entity/other/cavalry/CavalryHorseEntity.java

```diff
@@ -0,0 +1,876 @@
+        AttributeInstance jumpAttr = vanilla.getAttribute(Attributes.JUMP_STRENGTH);
+
+        // TODO: Create research that improves the capability of CavalryHorses
+        double maxHealth = healthAttr != null ? healthAttr.getBaseValue() * 1.25 : 30.0D;
+        double moveSpeed = speedAttr != null ? speedAttr.getBaseValue() * 1.25 : 0.25D;
+        double jumpStrength = jumpAttr != null ? 0.7D : 0.7D;
+
+        // --- Snapshot Horse-specific state (variant/armor) if applicable ---
+        Variant variant = null;
+        if (vanilla instanceof Horse h)
+        {
+            variant = h.getVariant();
+        }
+
+        // Leash (if any)
+        Entity leashHolder = vanilla.getLeashHolder();
+
+        // Convert to our entity type ---
+        CavalryHorseEntity cav = vanilla.convertTo(ModEntities.CAVALRY_HORSE, true);
+        if (cav == null) return null;
+
+        // Re-apply attributes & health
+        AttributeInstance cavHealthAttr = cav.getAttribute(Attributes.MAX_HEALTH);
+        if (cavHealthAttr != null)
+        {
+            cavHealthAttr.setBaseValue(maxHealth);
+        }
+
+        AttributeInstance cavSpeedAttr = cav.getAttribute(Attributes.MOVEMENT_SPEED);
+        if (cavSpeedAttr != null)
+        {
+            cavSpeedAttr.setBaseValue(moveSpeed);
+        }
+
+        AttributeInstance cavJumpAttr = cav.getAttribute(Attributes.JUMP_STRENGTH);
+        if (cavJumpAttr != null)
+        {
+            cavJumpAttr.setBaseValue(jumpStrength);
+        }
+        cav.setMaxUpStep(1.1F);
+
+        cav.setHealth((float) Math.min(health, maxHealth));
+
+        // Re-apply AbstractHorse state
+        cav.setTamed(wasTamed);
+        cav.setOwnerUUID(owner);
+        cav.setTemper(temper);
+        cav.setPersistenceRequired();
+
+        // Re-apply Horse-specific visuals
+        if (variant != null)
+        {
+            cav.setVariant(variant);
+        }
+
+        // Name & leash
+        if (customName != null)
+        {
+            cav.setCustomName(Component.literal(customName));
+        }
+
+        if (leashHolder != null)
+        {
+            cav.setLeashedTo(leashHolder, true);
+        }
+
+        return cav;
+    }
+
+    /**
+     * Whether this entity should be saved to disk.
+     * <p>As CavelryHorse entities are always saved to disk, this method always returns true.
+     */
+    @Override
+    public boolean shouldBeSaved()
+    {
+        return true;
+    }
+
+    /**
+     * Applies damage to this entity.
+     * A portion of that damage is also applied to the combat cooldown counter.
+     *
+     * @param damageSource the source of the damage
+     * @param damageAmount the amount of damage to apply
+     * @return true if the damage was applied, false otherwise
+     */
+    @Override
+    public boolean hurt(@Nonnull DamageSource damageSource, float damageAmount)
+    {
+        // TODO: Create research that provides combat cooldown mitigation
```

Comment:
we also need to decide whether horse and rider are one unit togher and die together, or if the horse can die and the guard dismounts(horse does take damage first then?)